### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by E-Sharpilo**
[DEMO LINK](https://E-Sharpilo.github.io/react_add-todo-form/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,19 +1,162 @@
...
+              && (
+                <span className="error">
+                  Please choose a user
+                </span>
+              )}
+          </label>
+          <button
+            type="submit"
+            className="button"
+            onClick={this.addTask}
```
Comment (aliceblunt8) [inline src/App.tsx:147]: add todo on submit action on the form

File: src/App.tsx [hunk 2]
```diff
@@ -1,19 +1,162 @@
...
+          >
+            Add Task
+          </button>
+        </form>
+
+        <TodoList
+          todos={this.state.todos}
+          users={this.state.users}
```
Comment (aliceblunt8) [inline src/App.tsx:155]: destructure everything from state

File: src/App.tsx [hunk 3]
```diff
@@ -1,19 +1,162 @@
...
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <form
+          className="form"
+          action=""
+          method="GET"
+          onSubmit={e => (e.preventDefault())}
```
Comment (aliceblunt8) [inline src/App.tsx:91]: move e.preventDefault inside addTask

```suggestion
          onSubmit={this.addTask}
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,19 +1,162 @@
...
+    };
+
+    return this.setState((state) => ({
+      todos: [...state.todos, newTodo],
+      title: '',
+      selectedUser: '',
+    }));
+  };
+
+  Error = () => {
```
Comment (aliceblunt8) [inline src/App.tsx:70]:
```suggestion
  showError = () => {
```

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

**Review by danylo-onishchuk — APPROVED** [review]

**Pull request closed without merging**